src/session/isAuthenticated.js

```diff
@@ -8,8 +11,10 @@ import { getAccessToken } from "../utils/getAccessToken";
  * @returns {() => Promise<boolean>}
  */
 export const isAuthenticatedFactory = (req, res) => async () => {
-  const token = getAccessToken(req, res);
-
+  const token = await getAccessToken(req, res);
+  if(isTokenExpired(token)) {
+    redirect('/api/auth/login')
+  }
```

Comment:
This is by design - cookies cannot be modified in React Server Components, so to prevent edge cases where stale data may be accessed in an RSC _outside_ of middleware, we redirect on expiry.

This isn't too jarring as the login should be instant (given the refresh token is still valid)